commands/generate-key.js

```diff
@@ -13,8 +13,11 @@ module.exports = {
             if (argv.accountId) {
                 console.log('WARN: Account id is provided but ignored in case of using Ledger.');
             }
-            const publicKey = await argv.signer.getPublicKey();
-            // NOTE: Command above already prints public key.
+            console.log(`Please, confirm on the Ledger receiveng of the public key for HD path ${argv.useLedgerKey}`);
+            const publicKey = await argv.signer.getPublicKey(false);
+            if (!publicKey) return;
+            console.log('Please, confirm that this key is the one that is displayed on the Ledger screen now');
+            if (!await argv.signer.getPublicKey(false)) return;
```

Comment:
not sure I understand why we call `signer.getPublicKey(false)` 2 times in a row

Reply:
@vgrichina this is the point of this PR. We want to give users the opportunity to compare what is printed in the terminal and what is shown on the ledger screen. That is why we need to call `getPublicKey` two times. More details in the attached issue.

Reply:
So user basically would have to approve the key twice on Ledger, correct?

I think it's a bit suboptimal, we probably need to change app itself to work differently. For example on first call ask user to confirm displayed HD path and return public key and on a second call just ask user to confirmed displayed public key.
So basically have 2 diff APIs on Ledger:
1) show HD path, return public key if confirmed
2) display public key for HS path (just for user to confirm it's the same)

Above not blocking this PR however. Maybe makes sense to add `--yolo` option which won't ask for extra confirmation (useful during development for sure and main users are developers here).

Reply:
@vgrichina I have added `--yolo` option and created a separate issue for Ledger here: https://github.com/near/near-ledger-js/issues/42
I would say, that the only way to be sure in you PK and HD path is to show them on the screen at the same time. Two times, since we still need to compare them. In this case, we will need only 1 function that will show the next info: `HD path: path PK: public key` -> [OK]

Reply:
so there are 2 aspects here:
1) you want to avoid using wrong public key somewhere because near-cli is compromised (this one you already solved)
2) you want to avoid random apps getting public key from Ledger without user consent (that's why you kinda have to ask on Ledger before near-cli displays key)